### DISCUSSION TITLE
ISO file reading

### DISCUSSION THREAD

**Discussion by chenlinjie1213** [firemodels/fds#11105] [General]
Dear all, I would like to ask how to open and edit the ISO file exported by FDS calculation? I tried to open the ISO file with winrar software, but it shows that the ISO file is corrupted. Can you tell me how to edit the ISO file? Thank you!

**Comment by gforney** [comment]
The format used to output isosurface, .iso files is documented in.an
appendix in the fds users guide.

**Comment by chenlinjie1213** [comment]
Yes, thank you very much for your reply




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "firemodels/fds"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月12日(星期六) 凌晨0:15
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [firemodels/fds] ISO file reading (Discussion #11105)





 The format used to output isosurface, .iso files is documented in.an
 appendix in the fds users guide.
 
 On Fri, Nov 11, 2022, 11:04 AM chenlinjie1213 ***@***.***&gt;
 wrote:
 
 &gt; Dear all, I would like to ask how to open and edit the ISO file exported
 &gt; by FDS calculation? I tried to open the ISO file with winrar software, but
 &gt; it shows that the ISO file is corrupted. Can you tell me how to edit the
 &gt; ISO file? Thank you!
 &gt;
 &gt; —
 &gt; Reply to this email directly, view it on GitHub
 &gt; <https://github.com/firemodels/fds/discussions/11105&gt;, or unsubscribe
 &gt; <https://github.com/notifications/unsubscribe-auth/AC6UCRVTJUT5UY4VSITJZVLWHZVBZANCNFSM6AAAAAAR5XKEXI&gt;
 &gt; .
 &gt; You are receiving this because you are subscribed to this thread.Message
 &gt; ID: ***@***.***&gt;
 &gt;

**Comment by chenlinjie1213** [comment]
Dear professor, I may need your help, thank you. We want to read the relevant plot3D sequence data and then do the relevant data analysis, but I'm encountering obstacles. 


For example, we want to read the **plot3D data** of temperature, currently, if we take the toolkit developed by FDS team, we can only export the .dat data set at a certain moment by running fds2ascii.exe once, how can we run fds2ascii.exe once to export all the .dat files in the numerical simulation time (according to one second time interval to export a .dat file)? 

Thank you very much.

**Comment by mcgratta** [comment]
fds2ascii was developed to analyze a single Plot3d  file. For multiple files, you either have a write a script of some sort to invoke fds2ascii multiple times, or you can rework the fds2ascii.f90 code to do whatever you want.

**Comment by chenlinjie1213** [comment]
OK, thank you.


Thanks again


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "firemodels/fds"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年12月14日(星期三) 晚上10:33
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [firemodels/fds] ISO file reading (Discussion #11105)





 
fds2ascii was developed to analyze a single Plot3d  file. For multiple files, you either have a write a script of some sort to invoke fds2ascii multiple times, or you can rework the fds2ascii.f90 code to do whatever you want.

**Comment by chenlinjie1213** [comment]
Dear&nbsp;professor , we want to ask if the visibility calculation in the result file in Pyrosim has considered the shielding effect of smoke on the light?


In addition, we also want to ask a question: Is the visibility in Pyrosim measured under what intensity of illumination?&nbsp;Because the influence of smoke on visibility may be more slight when the illumination is high.

&nbsp;
Thanks a lot.






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "firemodels/fds"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年12月14日(星期三) 晚上10:33
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [firemodels/fds] ISO file reading (Discussion #11105)





 
fds2ascii was developed to analyze a single Plot3d  file. For multiple files, you either have a write a script of some sort to invoke fds2ascii multiple times, or you can rework the fds2ascii.f90 code to do whatever you want.

**Comment by mcgratta** [comment]
Visibility is based solely no Beer's Law; that is, the absorption of light is proportional to the concentration of smoke or I = I0 exp(-K L) where I is the light intensity, K the absorption coefficient and L the distance.

**Comment by chenlinjie1213** [comment]
Dear professor,
THANKS A LOT.​​

In addition, there is another question.
When FDS calculates visibility, is it measured under what lighting conditions?&nbsp;(Is the default visibility of the FDS software 30m, and then with the increasing concentration of smoke, does the visibility decrea​se?)​​​​


Thanks again.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "firemodels/fds"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月11日(星期六) 晚上11:56
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [firemodels/fds] ISO file reading (Discussion #11105)





 
Visibility is based solely no Beer's Law; that is, the absorption of light is proportional to the concentration of smoke or I = I0 exp(-K L) where I is the light intensity, K the absorption coefficient and L the distance.

**Comment by drjfloyd** [comment]
Visbiilty is Eq. 21.22 in the User's Guide

S = C / K where K = Km rho Y_S

The limit of S as Y_S goes to zero is infinity. We have chosen to limit the upper value of visiiblity to MAXIMUM_VISIBILITY which has a default value of 30 m. This way when you load a slice file of visibility in smokeview, you get a useful colorbar scale for the duration of a simulation. If we set MAXIMUM_VISIBILITY=10,000, then you wouldn't have any resolution of low visibility vlaues that begin to impact egress.

There is no explicit assumption on lighting condition; however, there is an implicit one. The values for C come from human experiments perceiving lit or unlit objects in smokey environments. There is a range of values for each condition that resulted from various experiments and 3 and 8 are the midpoints of those ranges. You would have to look at the original research that measured those values to see what the lighting conditions were (see the SFPE Handbook discussion on visibility for a list of references).

**Comment by chenlinjie1213** [comment]
THANKS A LOT.




This reply is very valuable and helpful.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "firemodels/fds"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月12日(星期天) 上午9:50
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [firemodels/fds] ISO file reading (Discussion #11105)





 
Visbiilty is Eq. 21.22 in the User's Guide
 
S = C / K where K = Km rho Y_S
 
The limit of S as Y_S goes to zero is infinity. We have chosen to limit the upper value of visiiblity to MAXIMUM_VISIBILITY which has a default value of 30 m. This way when you load a slice file of visibility in smokeview, you get a useful colorbar scale for the duration of a simulation. If we set MAXIMUM_VISIBILITY=10,000, then you wouldn't have any resolution of low visibility vlaues that begin to impact egress.
 
There is no explicit assumption on lighting condition; however, there is an implicit one. The values for C come from human experiments perceiving lit or unlit objects in smokey environments. There is a range of values that resulted from various experiments and 3 and 8 are the midpoints of that range. You would have to look at the original research that measured those values to see what the lighting conditions were (see the SFPE Handbook discussion on visibility for a list of refereces).

**Comment by chenlinjie1213** [comment]
Dear engineer, I would like to ask if the FDS currently developed by your team has any GIS related development interface? Thank you.


Currently, I would like to import the .sf file from the calculation result file in FDS to GIS for display and then do further development, but unfortunately, we cannot read the .sf file through GIS at the moment.


Thanks again and good luck!





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "firemodels/fds"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月11日(星期六) 晚上11:56
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [firemodels/fds] ISO file reading (Discussion #11105)





 
Visibility is based solely no Beer's Law; that is, the absorption of light is proportional to the concentration of smoke or I = I0 exp(-K L) where I is the light intensity, K the absorption coefficient and L the distance.

**Comment by emanuelegissi** [comment]
Currently the only visualization interface is Smokeview. But FDS file
formats are open and fully documented, so you can write a conversion script
to another format.

Il dom 19 feb 2023, 04:11 chenlinjie1213 ***@***.***> ha
scritto:

**Comment by chenlinjie1213** [comment]
ok, thank you




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "firemodels/fds"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月19日(星期天) 晚上9:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [firemodels/fds] ISO file reading (Discussion #11105)





 Currently the only visualization interface is Smokeview. But FDS file
 formats are open and fully documented, so you can write a conversion script
 to another format.
 
 Il dom 19 feb 2023, 04:11 chenlinjie1213 ***@***.***&gt; ha
 scritto:
 
 &gt; Dear engineer, I would like to ask if the FDS currently developed by your
 &gt; team has any GIS related development interface? Thank you.
 &gt;
 &gt;
 &gt; Currently, I would like to import the .sf file from the calculation result
 &gt; file in FDS to GIS for display and then do further development, but
 &gt; unfortunately, we cannot read the .sf file through GIS at the moment.
 &gt;
 &gt;
 &gt; Thanks again and good luck!
 &gt;
 &gt;
 &gt;
 &gt;
 &gt;
 &gt; ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------
 &gt; 发件人: "firemodels/fds" ***@***.***&amp;gt;;
 &gt; 发送时间:&amp;nbsp;2023年2月11日(星期六) 晚上11:56
 &gt; ***@***.***&amp;gt;;
 &gt; ***@***.******@***.***&amp;gt;;
 &gt; 主题:&amp;nbsp;Re: [firemodels/fds] ISO file reading (Discussion #11105)
 &gt;
 &gt;
 &gt;
 &gt;
 &gt;
 &gt;
 &gt; Visibility is based solely no Beer's Law; that is, the absorption of light
 &gt; is proportional to the concentration of smoke or I = I0 exp(-K L) where I
 &gt; is the light intensity, K the absorption coefficient and L the distance.
 &gt;
 &gt; —
 &gt; Reply to this email directly, view it on GitHub, or unsubscribe.
 &gt; You are receiving this because you commented.Message ID: ***@***.***&amp;gt;
 &gt;
 &gt; —
 &gt; Reply to this email directly, view it on GitHub
 &gt; <https://github.com/firemodels/fds/discussions/11105#discussioncomment-5021878&gt;,
 &gt; or unsubscribe
 &gt; <https://github.com/notifications/unsubscribe-auth/ACXMVYYOQQSCBWLQ55GNZSDWYGFPBANCNFSM6AAAAAAR5XKEXI&gt;
 &gt; .
 &gt; You are receiving this because you are subscribed to this thread.Message
 &gt; ID: ***@***.***&gt;
 &gt;
